### ISSUE TITLE
Problem layers shape

### ISSUE THREAD

**Issue by kingjr** [adamcasson/c3d#2]
Hi,

First thanks for your package.

I'm encountering an issue when trying to use the pretrained C3D model. The output shape declared don't seem to systematically match the actual shape. 

For example, layer `pool1` is supposed to be of shape `(None, 16, 56, 56, 64)` but when I run the model it generate an output of shape `(1, 14, 57, 57, 64)`

The mistmaches aren't systematic, but can be easily checked with the following example.

Would you have any idea on why that is?

```python
import numpy as np
from keras.models import Model
from c3d import C3D
base_model = C3D(weights='sports1M')

# input of 16-images movie
x = np.zeros((1, 16, 112, 112, 3))

# for each layer, print expected and actual output shape
for layer in base_model.layers:
    head_model = Model(inputs=base_model.input,
                       outputs=layer.output)
    print(layer.name, 
          layer.output_shape,
          head_model.predict(x).shape)

```

outputs:
```
('conv1', (None, 16, 112, 112, 64), (1, 16, 112, 112, 64))
('pool1', (None, 16, 56, 56, 64), (1, 14, 57, 57, 64))
('conv2', (None, 16, 56, 56, 128), (1, 14, 57, 57, 128))
('pool2', (None, 8, 28, 28, 128), (1, 7, 28, 28, 128))
('conv3a', (None, 8, 28, 28, 256), (1, 7, 28, 28, 256))
('conv3b', (None, 8, 28, 28, 256), (1, 7, 28, 28, 256))
('pool3', (None, 4, 14, 14, 256), (1, 3, 14, 14, 256))
('conv4a', (None, 4, 14, 14, 512), (1, 3, 14, 14, 512))
('conv4b', (None, 4, 14, 14, 512), (1, 3, 14, 14, 512))
('pool4', (None, 2, 7, 7, 512), (1, 1, 7, 7, 512))
('conv5a', (None, 2, 7, 7, 512), (1, 1, 7, 7, 512))
('conv5b', (None, 2, 7, 7, 512), (1, 1, 7, 7, 512))
('zero_padding3d_1', (None, 2, 9, 9, 512), (1, 2, 9, 9, 512))
('pool5', (None, 1, 4, 4, 512), (1, 1, 4, 4, 512))
('flatten_1', (None, 8192), (1, 8192))
('fc6', (None, 4096), (1, 4096))
('dropout_1', (None, 4096), (1, 4096))
('fc7', (None, 4096), (1, 4096))
('dropout_2', (None, 4096), (1, 4096))
('fc8', (None, 487), (1, 487))
```

**Comment by kingjr** [comment]
I believe this is due to an issue related to the Conv3D padding: https://github.com/keras-team/keras/issues/9389

**Comment by kingjr** [comment]
close by keras-team/keras#9389
